pkg/kubelet/logs/container_log_manager.go

```diff
@@ -138,9 +138,6 @@ func parseMaxSize(size string) (int64, error) {
 	if !ok {
 		return 0, fmt.Errorf("invalid max log size")
 	}
-	if maxSize < 0 {
-		return 0, fmt.Errorf("negative max log size %d", maxSize)
-	}
```

Comment:
https://github.com/kubernetes/kubernetes/blob/d937ba268a8ed814db74ee6b28f4cc7d4ea9e461/pkg/kubelet/logs/container_log_manager.go#L250

This value will be used here, and it is not allowed to be negative.

Reply:
I notice this value, it is ok for the negative value, and  I don't see where it's not allowed

Reply:
https://github.com/kubernetes/kubernetes/blob/d937ba268a8ed814db74ee6b28f4cc7d4ea9e461/pkg/kubelet/logs/container_log_manager.go#L64-L73
> Negative number means to disable container log rotation. 

If it is a negative number, it is actually not necessary to start this. maybe skip better.
If it is 0, is it using a default?

https://github.com/kubernetes/kubernetes/blob/d937ba268a8ed814db74ee6b28f4cc7d4ea9e461/pkg/kubelet/logs/container_log_manager.go#L174-L182

``` golang
// Start the container log manager. 
 func (c *containerLogManager) Start() { 
	if c.policy.MaxSize < 0 {
		return 
	}
 	// Start a goroutine periodically does container log rotation. 
 	go wait.Forever(func() { 
 		if err := c.rotateLogs(); err != nil { 
 			klog.Errorf("Failed to rotate container logs: %v", err) 
 		} 
 	}, logMonitorPeriod) 
 } 
```

Reply:
I get it. I will try to improve it